src/Raven.Server/Documents/Indexes/MapReduce/AutoMapReduceIndex.cs

```diff
@@ -93,25 +73,23 @@ public override IDisposable InitializeIndexingWork(TransactionOperationContext i
 
         public override unsafe void HandleDelete(DocumentTombstone tombstone, IndexWriteOperation writer, TransactionOperationContext indexContext, IndexingStatsScope stats)
         {
-            var etagSlice = new Slice((byte*)null, sizeof(long));
-
-            foreach (var mapEntry in GetMapEntriesForDocument(_indexingWorkContext.MapEntriesTable, tombstone.Key))
-            {
-                ReduceKeyState state;
-                if (_indexingWorkContext.StateByReduceKeyHash.TryGetValue(mapEntry.ReduceKeyHash, out state) == false)
-                {
-                    //TODO: Need better way to handle tree names
-                    var tree = indexContext.Transaction.InnerTransaction.CreateTree("TODO_" + mapEntry.ReduceKeyHash);
-                    _indexingWorkContext.StateByReduceKeyHash[mapEntry.ReduceKeyHash] = state = new ReduceKeyState(tree);
-                }
+            var mapEntry = GetMapEntryForDocument(_indexingWorkContext.MapEntriesTable, tombstone.Key);
 
-                var etag = mapEntry.Etag;
-                etagSlice.Set((byte*)&etag, sizeof(long));
+            if (mapEntry == null)
+                return;
 
-                state.Tree.Delete(etagSlice);
-
-                _indexingWorkContext.MapEntriesTable.Delete(mapEntry.StorageId);
+            ReduceKeyState state;
+            if (_indexingWorkContext.StateByReduceKeyHash.TryGetValue(mapEntry.ReduceKeyHash, out state) == false)
+            {
+                //TODO: Need better way to handle tree names
+                var tree = indexContext.Transaction.InnerTransaction.ReadTree("TODO_" + mapEntry.ReduceKeyHash);
+                _indexingWorkContext.StateByReduceKeyHash[mapEntry.ReduceKeyHash] = state = new ReduceKeyState(tree);
             }
+
+            var storageId = mapEntry.StorageId;
+ 
+            state.Tree.Delete(new Slice((byte*)&storageId, sizeof(long)));
+            _indexingWorkContext.MapEntriesTable.Delete(mapEntry.StorageId);
```

Comment:
Yeah, I will add appropriate comments to the code when we store / delete map-reduce stuff.